libs/langgraph/langgraph/utils.py

```diff
@@ -183,3 +189,20 @@ def coerce_to_runnable(thing: RunnableLike, *, name: str, trace: bool) -> Runnab
             f"Expected a Runnable, callable or dict."
             f"Instead got an unsupported type: {type(thing)}"
         )
+
+
+def is_optional_type(type_: Any) -> bool:
+    """Check if a type is Optional."""
+
+    if hasattr(type_, "__origin__") and hasattr(type_, "__args__"):
+        origin = get_origin(type_)
+        if origin is Optional:
+            return True
+        if origin is Union:
+            return any(
+                arg is type(None) or is_optional_type(arg) for arg in type_.__args__
+            )
+        return origin is None
+    if hasattr(type_, "__bound__") and type_.__bound__ is not None:
+        return is_optional_type(type_.__bound__)
```

Comment:
This currently does *not* type Any as optional, which maybe feels contradictory given that Optional[Any] equals Union[Any, None] and Any covers the None case?